### ISSUE TITLE
Resource mod_fastcgi update

### ISSUE THREAD

**Issue by damacus** [sous-chefs/apache2#575]
### Cookbook version
6.0.0-beta

### Chef-client version
Latest

### Platform Details
- All supported

### Scenario:
As a user
I want to be able to configure mod_fastcgi via properties rather than attributes
So that I can interact with mods in the same way.

### Expected Result:
- [ ] apache2_mod_fastcgi is configurable via resource properties
- [ ] mod_fastcgi has unit tests 
- [ ] no attributes under node['apache']['mod_fastcgi'] exist

**Comment by JohnRoesler** [comment]
I may be missing something, but the only `['apache']['mod_fastcgi']` attributes I'm seeing are in the rb for fastcgi relating to downloading the package [here](https://github.com/sous-chefs/apache2/blob/master/attributes/mod_fastcgi.rb#L21-L23). To that end, should the resource for fastcgi also download and compile like the recipe did [here](https://github.com/sous-chefs/apache2/blob/master/recipes/mod_fastcgi.rb#L33-L55)? I thought we had talked today about not wanting to be compiling. If that's the case, what's left to do, and how does an end user get the fastcgi package if not through the resource? Hopefully that made some sense 😄  Happy to hack on this one when I understand better.

**Comment by JohnRoesler** [comment]
From everything I'm finding, mainly the docs [here](https://httpd.apache.org/mod_fcgid/), the only available options are the source and instructions on compiling.

**Comment by damacus** [comment]
Maybe we can throw this into another cookbook. So people aren't having to dep on build tools, and put it in the readme

**Comment by JohnRoesler** [comment]
Yeah, I like that. I'm working on the fastcgi cookbook. Should have something soon!

**Comment by JohnRoesler** [comment]
Take a look -> https://github.com/JohnRoesler/apache2_mod_fastcgi/tree/initial

So WIP. Needs more tests. If this looks on the right path, what's the process to get it into sous-chefs? Do I PR the terraform and make the repo, then commit all this?
